### PR TITLE
Bump cache module to version 1.0.0

### PR DESCRIPTION
**Pull request by dhadka**
Use version 1.0.0 of the `@actions/cache` [module](https://www.npmjs.com/package/@actions/cache).  Cache files hosted on Azure will now use the Azure Storage SDK for NodeJS to download the files, replacing the [http-client](https://github.com/actions/http-client).

Also bumps the cache action version to 2.1.0.

### CODE REVIEW

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -52,7 +52,7 @@ jobs:
       shell: bash
       run: |
         npm run build
-        if [ "$(git status --porcelain | wc -l)" -gt "0" ]; then
+        if [ "$(git diff --ignore-space-at-eol | wc -l)" -gt "0" ]; then
```
Comment (dhadka) [inline .github/workflows/workflow.yml:55]: `tslib.js` in `@azure/core-auth` uses CRLF line endings which gets flagged by this step.  Ignoring EOL characters.

**Comment by dhadka** [comment]
This PR had two CodeQL warnings coming from the new Azure libraries.  Marked as false positive.  We were reading the AZURE_LOG_LEVEL env var and printing the value to the console.  As that env var is not a secret, it is not a security concern.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "cache",
-  "version": "1.1.2",
+  "version": "2.1.0",
```
Comment (aiqiaoy) [inline package.json:3]: Why are we bumping the cache action version?

Comment (dhadka) [reply package.json:3]: Sorry if I'm not understanding, but shouldn't this match the tag / release version?  I'm planning to release this change as `@v2.1`

**Pull request merged**